index.js

```diff
@@ -141,44 +192,60 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+let ticketNumber = 1
+function takeATicketNumber(line){
+  console.log(`Welcome. You are ticket number ${ticketNumber}`)
+  line.push(ticketNumber)
+  ticketNumber++
+}
+console.log(takeATicketNumber(line))
+// // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// // // => `Welcome. You are ticket number 2`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(nowServing(line))
+// // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// // // => `Currently serving 2.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(takeATicketNumber(line))
+// // // => `Welcome. You are ticket number 3`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
-
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+const newLine = []
+const takeATicketNumberFunc = ticketNumberGeneratorFunc()
+
+function ticketNumberGeneratorFunc() {
+  let newTicketNumber = 1
+
+  return function takeATicketNumber2 (newLine){
+    console.log(`Welcome. You are ticket number ${newTicketNumber}`)
+    newLine.push(newTicketNumber)
+    newTicketNumber++
+  }  
+}
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumberFunc(newLine))
```

Comment:
Great job using the global variable approach and then being able to refactor that to use a closure. 👏